Source/FullScreenPokemon.ts

```diff
@@ -1403,6 +1403,72 @@ module FullScreenPokemon {
             }
         }
 
+        /**
+         * Starts the Player fishing.
+         *
+         * @param player   A Player to start fishing.
+         * @todo Add a parameter and functionality to handle the different rods.
+         */
+        startFishing(player: IPlayer): void {
```

Comment:
This function is too big. Split up the relevant parts into separate functions: one for the failure case (can't do that here) and one for the success case (can do that here).